src/BuildLoop.cpp

```diff
@@ -82,35 +74,32 @@ static void LogFirstTimeEnqueue(MemAllocLinear* scratch, RuntimeNode* enqueuedNo
     LogStructured(&msg);
 }
 
-static int EnqueueNodeListWithoutWakingAwaiters(BuildQueue* queue, MemAllocLinear* scratch, const FrozenArray<int32_t>& nodesToEnqueue, RuntimeNode* enqueingNode)
+static int EnqueueNodeListReversedWithoutWakingAwaitersOnlyWhenNotEnqueuedBefore(BuildQueue* queue, MemAllocLinear* scratch, const FrozenArray<int32_t>& nodesToEnqueue, RuntimeNode* enqueingNode)
```

Comment:
Should we use a shorter function name, and give it a good comment?

Reply:
Personally I would prefer not to. Comments are not seen at the call site when reading code. But I wil let @lucasmeijer have a say here since it's his function ;)

Reply:
The name is long, but I think it's warranted here.   the behaviour described in the function name is important to understand from the callsite and the function body.